dialer.go

```diff
@@ -526,7 +526,7 @@ func matchSelectedExtensions(selected []byte, wanted, received []httphead.Option
 // connection "poisoned" by SetDeadline() call. In that case done(&err) will
 // store at *err ctx.Err() result. If err is caused not by timeout, it will
 // leaved untouched.
-func setupContextDeadliner(conn net.Conn, ctx context.Context) (done func(*error)) {
+func setupContextDeadliner(ctx context.Context, conn net.Conn) (done func(*error)) {
```

Comment:
Position of args was chosen consciously – context has no impact on the operation of `setupContextDeadliner` – it controls the lifetime of `conn`.

Reply:
It should be interpreted like if it will be the `time.Duration` type or `<-chan struct{}`.

Reply:
I did this reorder just because it is written in go best practices